server/src/main/java/org/opensearch/search/internal/ReaderContext.java

```diff
@@ -66,9 +66,9 @@ public class ReaderContext implements Releasable {
     private final boolean singleSession;
 
     private final AtomicLong keepAlive;
-    private final AtomicLong lastAccessTime;
+    protected final AtomicLong lastAccessTime;
```

Comment:
Please hide those behind the method call, like `tryUpdateKeepAlive` (`tryUpdateLastAccessTime`, ...) instead of relaxing visibility.